lib/SILGen/LValue.h

```diff
@@ -405,7 +405,11 @@ class LValue {
     assert(&component == Path.back().get());
     Path.pop_back();
   }
-  
+
+  /// Pop the last component off this LValue unsafely. Please be careful when
+  /// using this!
+  void unsafelyDropLastComponent() & { Path.pop_back(); }
```

Comment:
Can we check whether that last component is the expected ownership component when we drop it?